src/js/facility-locator/components/Select.jsx

```diff
@@ -0,0 +1,160 @@
+    if (isToggle(e, isActive)) {
+      if (isActive && isSelect(e.keyCode)) {
+        this.handleFilterSelect(which.id, type);
+      }
+      return this.toggleDropdown(e, type);
+    }
+    if (isActive && isTraverse(e)) {
+      const increment = getDirection(e.keyCode);
+      const newIndex = this.state.focusedIndex + increment;
+      return this.focusOption(this.options[newIndex], newIndex, type);
+    }
+    return false;
+  }
+
+  handleBlur = (e) => {
+    const { currentTarget } = e;
+    setTimeout(() => {
+      if (!currentTarget.contains(document.activeElement)) {
+        this.setState({ dropdownActive: false });
+      }
+    });
+  }
+
+  toggleDropdown = (e) => {
+    const type = this.props.optionType;
+    const isActive = this.state.dropdownActive;
+    const { currentQuery: { facilityType, serviceType } } = this.props; // eslint-disable-line no-unused-vars
+    const queryType = type === 'service' ? serviceType : facilityType;
+    if (type === 'service' && noServices(type, facilityType)) {
+      return;
+    }
+    // To handle unrelated key events
+    if (e) {
+      const shouldNotToggle = e.keyCode && !isToggle(e, isActive);
+      if (shouldNotToggle) return;
+    }
+    if (isActive) {
+      this.dropdown.focus();
+    } else {
+      const { selection, id } = getSelection(this.options, queryType);
+      this.focusOption(selection, id, type);
+    }
+    this.setState({
+      dropdownActive: !this.state.dropdownActive
+    });
+  }
+
+  focusOption = (option, index) => {
+    if (option) {
+      option.focus();
+    }
+    if (this.options[index]) {
+      this.setState({
+        focusedIndex: index
+      });
+    }
+  }
+
+  render() {
+    const { optionType, currentQuery, isMobile, hasIcons } = this.props;
+    const { facilityType, serviceType } = currentQuery;
+    const disabled = optionType === 'service' && !['benefits', 'vet_center'].includes(facilityType);
+    let queryType = optionType === 'service' ? serviceType : facilityType;
+    if (!queryType) {
+      if (optionType === 'service') {
+        queryType = 'All';
+      } else if (optionType === 'facility') {
+        queryType = 'AllFacilities';
+      }
+    }
+    const titleType = optionType[0].toUpperCase() + optionType.slice(1);
+    const dropdownClasses = classNames({
+      'facility-dropdown-wrapper': true,
+      active: this.state.dropdownActive,
+      disabled
+    });
+    return (
+      <div onBlur={this.handleBlur} className="columns usa-width-one-fourth medium-3">
+        <label htmlFor={`${optionType}Type`} id={`${optionType}-label`}>{`Select ${titleType} Type`}</label>
+        <button
+          onKeyDown={e => this.navigateDropdown(e, optionType)}
+          className={dropdownClasses}
+          ref={elem => { this.dropdown = elem; }}
+          aria-expanded="false"
+          aria-labelledby={`${optionType}-label`}
+          onClick={e => {e.preventDefault(); this.toggleDropdown(e);}}>
+          <Listbox
```

Comment:
Is nesting these options inside of a button ok, a11y-wise? I'm curious how you ended up with that structure.

Reply:
Yes, but buttons are not allowed in buttons. This structure is based on Cauldron's [model](https://pattern-library.dequelabs.com/components/selects) but updated so that it conforms with the eslint-plugin-jsx-a11y no-noninteractive-element-interactions rule.

Reply:
It still seems slightly different than that, plus I don't see anything in that model that would run counter to that eslint rule. You can put click handlers on div elements. Was there a reason for choosing the button over a div and the combobox role?

Reply:
Good catch, this is now a div, the combobox role would cause the following violation here: Required ARIA child role not present: textbox (as it also does for Cauldron's model above)

Reply:
Actually adding the requisite tabIndex to the div causes a violation: `tabIndex` should only be declared on interactive elements
This was actually the reason a button was used. I will add the button role to the new div.

Reply:
Hm, I think this caused an error which was why I made it a button:
  - sitemap 1/4 (1m 50s)
   Failed [fail]: ('http://localhost:3333/facilities/: [critical] Certain ARIA roles must contain particular children\nSee https://dequeuniversity. undefined undefined)
   See https://dequeuniversity.com/rules/axe/2.3/aria-required-children?application=axeAPI
   
   <ul class="dropdown" role="listbox"></ul>
   #react-root > div > .row > .facility-locator > div > .desktop-container > div:nth-child(1) > .search-controls-container.clearfix > form > div:nth-child(3) > [role="button"] > div > ul[role="listbox"]"
       at /application/test/util/nightwatch-commands/axeCheck.js:56:19
       at Array.forEach (native)
       at Object.<anonymous> (/application/test/util/nightwatch-commands/axeCheck.js:48:16)

I should better document these decisions next time.

Reply:
@cehsu There's actually a bug in aXe here. I made a comment about it on your pre-need e2e test PR

Reply:
Thanks for the context @jbalboni, indeed, this seems the inverse of issues 214, 618 on the axe-core repo (which relate to the erroneous requirement that a specific child role be added when it shouldn't need to be), whereas in this case, it appears to be disallowing the assignment of child roles (listbox) to the button role (as the listbox children all posses the requisite option role -- so this is how I am interpreting the error). I do not see any issues related to this on the axe-core repo, do you propose we open a a bug issue on their repo and disable the check here for now?

Reply:
@cehsu I'm not sure what you mean, the text you pasted in above is the message we've been getting elsewhere because of the bug in 618.

Reply:
Thanks @jbalboni I will make use of the workaround you shared in that case.